VideoWeb/VideoWeb/ClientApp/src/app/services/events-hub.service.ts

```diff
@@ -0,0 +1,213 @@
+        return this.createConnectionBuilder()
+            .configureLogging(signalR.LogLevel.Debug)
+            .withAutomaticReconnect(this.reconnectionTimes)
+            .withUrl(eventHubPath, {
+                accessTokenFactory: () => this.oidcSecurityService.getToken()
+            })
+            .build();
+    }
+
+    configureConnection() {
+        this.connection.serverTimeoutInMilliseconds = this.serverTimeoutTime;
+        this.connection.onreconnecting(error => this.onEventHubReconnecting(error));
+        this.connection.onreconnected(() => this.onEventHubReconnected());
+        this.connection.onclose(error => this.onEventHubErrorOrClose(error));
+
+        this.eventsHubReady.next();
+    }
+
+    start() {
+        if (this.isWaitingToReconnect) {
+            this.logger.info('[EventsService] - A reconnection promise already exists');
+            return;
+        }
+
+        if (!this.isConnectedToHub) {
+            this.oidcSecurityService.isAuthenticated$.subscribe(authenticated => {
+                if (authenticated) {
+                    this._reconnectionAttempt++;
+                    this.connection
+                        .start()
+                        .then(() => {
+                            this.logger.info('[EventsService] - Successfully connected to EventHub');
+                            this._reconnectionAttempt = 0;
+                        })
+                        .catch(async error => {
+                            this.logger.warn(`[EventsService] - Failed to connect to EventHub ${error}`);
+                            this.onEventHubErrorOrClose(error); // TEST I THINK THIS IS REDUNDANT
+                            this.reconnect();
+                        });
+                } else {
+                    this.logger.debug(`[EventsService] - Cannot start - user is not authenticated`);
+                }
+            });
+        } else {
+            this.logger.debug(`[EventsService] - Cannot start - already connected to the event hub`);
+        }
+    }
+
+    reconnect() {
+        if (this.reconnectionTimes.length >= this.reconnectionAttempt) {
+            const delayMs = this.reconnectionTimes[this.reconnectionAttempt - 1];
+            this.logger.info(`[EventsService] - Reconnecting in ${delayMs}ms`);
+
+            this.reconnectionPromise = this.delay(delayMs).then(() => {
+                this.reconnectionPromise = null;
+                this.start();
+            });
+        } else {
+            this.logger.info(`[EventsService] - Failed to connect too many times (#${this.reconnectionAttempt}), going to service error`);
+            this.errorService.goToServiceError('Your connection was lost');
+
+            // Only subscibe to the first emitted event where the reconnection was successful
+            this.errorService.onUserTriggeredReconnect
+                .pipe(
+                    filter(reconnectionSuccessful => reconnectionSuccessful),
```

Comment:
Little shorthand: `filter(Boolean)`
It'll pass the value to Boolean's ctor, which returns true/false based on the constructed bool